### PR TITLE
feat: support start stop commands for multilevel switch

### LINKED ISSUE #30: [feat] Ability to send a stop value to the same topic as position request
Hi,

Im am slowly integrating into openhab, i want to use the openhab model "RollerShutter" to control my shutters.

It has 3 buttons -> up (value 99) -> stop ( value stop / or custom) -> down (value 0), i can also send a level in the form of a number between 0 - 100 (which is than mapped the openhab mqtt config to 0-99).

What is does is sending a request to the ```zwavejs2mqtt/study/switch_multilevel/endpoint_1/targetValue/set```

In there model i cannot select a other topic to send the stop command to.
It would be nice that i can send the "stop" word or any other value op the targetValue/set topic to use the build in openhab model, i think home assistant is using the same approach.

openhab

Currenly i am using the "dimmer" model to send values and use a openhab switch to send stop (by means of sending false) the Up or Down topic.

Thnx!

### PR DESCRIPTION
**Pull request by robertsLando**
Fixes #30

### CODE REVIEW

**Comment by ODINServ** [comment]
Hi,

I've tested it but it does not work but it does respond with an error.
I don't have HASS integration enabled, do i need to enablet hat?

The request: ```zwavejs2mqtt/maarten_left/switch_multilevel/endpoint_1/targetValue/set``` value: ```STOP``` or ```stop```, which seems to be translated to ```stopLevelChange```
```
 z2m:Mqtt Message received on zwavejs2mqtt/maarten_left/switch_multilevel/endpoint_1/targetValue/set +2s
  z2m:Zwave Error while writing stop on 5-38-1-targetValue: stopLevelChange failed +2s
  z2m:Zwave Unable to write stop on 5-38-1-targetValue +0ms
```
I've enabled HASS discovery but that seems not to matter.

Thnx!

**Comment by robertsLando** [comment]
`Stoplevelchange failed` is the error from zwavejs, it means that "stopLevelChange" has thrown, @AlCalzone do I made something wrong?

File: lib/ZwaveClient.js [hunk 1]
```diff
@@ -1899,12 +1904,33 @@ ZwaveClient.prototype.writeValue = async function (valueId, value) {
...
-      result = await this.getNode(valueId.nodeId).setValue(valueId, value)
+      const zwaveNode = await this.getNode(valueId.nodeId)
+
+      // handle multilevel switch 'start' and 'stop' commands
+      if (
+        valueId.commandClass === CommandClasses['Multilevel Switch'] &&
+        isNaN(value)
+      ) {
+        if (/stop/i.test(value)) {
+          await zwaveNode.commandClasses['Multilevel Switch'].stopLevelChange()
```
Comment (robertsLando) [inline lib/ZwaveClient.js:1924]: @AlCalzone here is where I call it.

Comment (AlCalzone) [reply lib/ZwaveClient.js:1924]: Do you get any specific error? Lots of things can go wrong.

Comment (robertsLando) [reply lib/ZwaveClient.js:1924]: `stopLevelChange failed` that's the error :man_shrugging:

Comment (ODINServ) [reply lib/ZwaveClient.js:1928]: Ive attached logging to the issue: https://github.com/zwave-js/zwavejs2mqtt/issues/30#issuecomment-736558562

```
14:31:50.571 DRIVER » [Node 004] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      73
                      └─[SupervisionCCGet]
                        │ session id:      10
                        │ request updates: false
                        └─[MultilevelSwitchCCStopLevelChange]
```

Comment (robertsLando) [reply lib/ZwaveClient.js:1928]: I don't think I have made something wrong becasue from the logs seems the call is done correctly but something else has failed

Comment (robertsLando) [reply lib/ZwaveClient.js:1928]:
```
14:31:50.571 DRIVER » [Node 004] [REQ] [SendData]
                      │ transmit options: 0x25
                      │ callback id:      73
                      └─[SupervisionCCGet]
                        │ session id:      10
                        │ request updates: false
                        └─[MultilevelSwitchCCStopLevelChange]
14:31:50.571 DRIVER   send thread state: sending.execute
14:31:50.574 SERIAL « [ACK]                                                                   (0x06)
14:31:50.575 DRIVER   send thread state: sending.execute
14:31:50.578 SERIAL « 0x0104011301e8                                                       (6 bytes)
14:31:50.578 SERIAL » [ACK]                                                                   (0x06)
14:31:50.579 DRIVER « [RES] [SendData]
                        was sent: true
14:31:50.581 DRIVER   send thread state: sending.execute
14:31:50.593 SERIAL « 0x0107001349000002a0                                                 (9 bytes)
14:31:50.594 SERIAL » [ACK]                                                                   (0x06)
14:31:50.595 DRIVER « [REQ] [SendData]
                        callback id:     73
                        transmit status: OK
14:31:50.598 DRIVER   send thread state: sending.execute
14:31:50.598 DRIVER   send thread state: sending.waitForUpdate
14:31:50.608 SERIAL « 0x010b00040004056c020a000095                                        (13 bytes)
14:31:50.609 SERIAL » [ACK]                                                                   (0x06)
14:31:50.611 DRIVER « [Node 004] [REQ] [ApplicationCommand]
                      └─[SupervisionCCReport]
                          session id:          10
                          more updates follow: false
                          status:              NoSupport
                          duration:            [Duration: 0seconds]
```

>  status:              NoSupport

Not supported from the device?

Comment (AlCalzone) [reply lib/ZwaveClient.js:1928]: I found it:
```
14:31:50.611 DRIVER « [Node 004] [REQ] [ApplicationCommand]
                      └─[SupervisionCCReport]
                          session id:          10
                          more updates follow: false
                          status:              NoSupport
                          duration:            [Duration: 0seconds]
```
where `NoSupport` means "The command is not supported by the receiver."

Comment (AlCalzone) [reply lib/ZwaveClient.js:1928]: Which is weird because `MultilevelSwitchCCStopLevelChange` is a V1 command.

Comment (AlCalzone) [reply lib/ZwaveClient.js:1928]: We could try opting out of using supervision, but I'd like to test if this is the problem first.

@robertsLando Can you try to send the command without going through the API:
```ts
		const cc = new MultilevelSwitchCCStopLevelChange(this.driver, {
			nodeId: /* The node's ID */,
			endpoint: /* The endpoint (or 0) */,
		});

		await this.driver.sendCommand(cc);
```

Comment (robertsLando) [reply lib/ZwaveClient.js:1928]: @AlCalzone Where do I found that `MultilevelSwitchCCStopLevelChange`?

Comment (robertsLando) [reply lib/ZwaveClient.js:1928]: General thought: from what I have undestanded most times do 'support checkes' seems isn't working most times even if it seems correct from a logical point of view. What about skip this kind of check and simply do things and return an error if they fails?

Comment (AlCalzone) [reply lib/ZwaveClient.js:1928]: Except the support check happens on the device in this case. It supports Supervision and responds to our request, telling us that `MultilevelSwitchCCStopLevelChange` is not supported (which I don't believe).

> What about skip this kind of checka and simply do things and return an error if they fails?

No. Using an unsupported command means we have to wait seconds to be sure that there is no response. If https://github.com/zwave-js/node-zwave-js/issues/997 happens, using an unsupported command means we'll wait 10 seconds when doing that.

Comment (AlCalzone) [reply lib/ZwaveClient.js:1933]: > Where do I found that MultilevelSwitchCCStopLevelChange?

Doesn't VSCode offer you an auto-import? Should be exported from `zwave-js/CommandClass`

Comment (robertsLando) [reply lib/ZwaveClient.js:1933]: > Doesn't VSCode offer you an auto-import? Should be exported from zwave-js/CommandClass

Nope, cannot find it, are you sure it's `zwave-js/CommandClass`?

Comment (robertsLando) [reply lib/ZwaveClient.js:1933]: > Doesn't VSCode offer you an auto-import? Should be exported from zwave-js/CommandClass

Nope, dunno why :man_shrugging: Anyway I have done now

Comment (robertsLando) [reply lib/ZwaveClient.js]: @ODINServ COuld you try now?

Comment (ODINServ) [reply lib/ZwaveClient.js]: Got a different error:
```
  z2m:Mqtt Message received on zwavejs2mqtt/maarten_right/switch_multilevel/endpoint_1/targetValue/set +7s
  z2m:Zwave Error while writing stop on 4-38-1-targetValue: MultilevelSwitchCCStopLevelChange is not a constructor +4s
  z2m:Zwave Unable to write stop on 4-38-1-targetValue +0ms
```

Comment (robertsLando) [reply lib/ZwaveClient.js]: :exploding_head:

Comment (robertsLando) [reply lib/ZwaveClient.js]: @AlCalzone ?

Comment (AlCalzone) [reply lib/ZwaveClient.js]: Ah shit, it is not exported from there. Try that path `zwave-js/build/lib/commandclass/MultilevelSwitchCC`.

Comment (robertsLando) [reply lib/ZwaveClient.js]: Fixed

Comment (robertsLando) [reply lib/ZwaveClient.js]: @ODINServ Could you try now

Comment (ODINServ) [reply lib/ZwaveClient.js]: It works!! :smiley: 

Your are my heroes! :muscle: 
I found in the mean time more users that want this, so your there heroes as well.
(https://community.home-assistant.io/t/fibaro-roller-shutter-3-fgr-223-cannot-get-it-to-work-properly/97577/115 )

:beers: :clinking_glasses: 

If you are interested in the logs: https://github.com/zwave-js/zwavejs2mqtt/issues/30#issuecomment-736645024

Comment (robertsLando) [reply lib/ZwaveClient.js]: @AlCalzone So based on this I think you will have to remove the check and just send the command? Can I revert to the previous version and use start/stopLevelChange?

Comment (AlCalzone) [reply lib/ZwaveClient.js]: Ok, I'll need to figure out why I used Supervision for those commands in the first place.

Comment (robertsLando) [reply lib/ZwaveClient.js]: @ODINServ start spam hass forums with zwavejs so :smile:

Comment (AlCalzone) [reply lib/ZwaveClient.js:1933]: Hmm, fuck past me.
I changed this [here](https://github.com/zwave-js/node-zwave-js/pull/648/files) but I no longer know why...

**Pull request merged**